utils/src/com/cloud/utils/Profiler.java

```diff
@@ -23,24 +23,20 @@
     private Long startTickInMs;
     private Long stopTickInMs;
 
-    public Profiler() {
-        startTickInMs = null;
-        stopTickInMs = null;
-    }
-
     public long start() {
-        startTickInMs = System.currentTimeMillis();
+        startTickInMs = System.nanoTime();
```

Comment:
The nanoTime() is given in Microseconds. So, the MS still fits in.

Cheers,
Wilder